src/clib/pio_darray.c

```diff
@@ -713,6 +715,22 @@ int PIOc_write_darray(int ncid, int varid, int ioid, PIO_Offset arraylen, void *
         needsflush = (maxfree <= 1.1 * (1 + wmb->num_arrays) * arraylen * iodesc->mpitype_size);
 #endif
 
+    /* When using PIO with PnetCDF + SUBSET rearranger the number
+       of non-contiguous regions cached in a single IO process can
+       grow to a large number. PnetCDF is not optimal at handling
```

Comment:
Minor edit : You might want to change "PnetCDF is not optimal"  to "PnetCDF is not efficient"